sympy/simplify/tests/test_simplify.py

```diff
@@ -806,3 +806,9 @@ def test_issue_15965():
     assert simplify(A + B) == anew + bnew
     assert simplify(A) == anew
     assert simplify(B) == bnew
+
+
+def test_issue_7971():
+    z = Integral(x, (x, 1, 1))
+    assert z is not S.Zero
+    assert simplify(z) == 0
```

Comment:
Shouldn't 0 here should be `S.Zero`?

Reply:
it doesn't matter: `0 == S.Zero`

Reply:
But it would be better to make this `is S.Zero` to affirm that the write short-circuit result is being sent back. If I make the change now, however, it will fail because of the Travis failure. I will make the change in another PR.